myriad-scheduler/src/main/java/com/ebay/myriad/state/SchedulerState.java

```diff
@@ -201,4 +228,58 @@ public MyriadState getMyriadState() {
     public boolean hasTask(Protos.TaskID taskID) {
         return this.tasks.containsKey(taskID);
     }
-}
\ No newline at end of file
+
+    public Protos.FrameworkID getFrameworkID() {
+         return this.frameworkId;
+    }
+
+    public void setFrameworkId(Protos.FrameworkID newFrameworkId) {
+        this.frameworkId = newFrameworkId;
+        updateStateStore();
+    }
+
+    public void updateStateStore() {
```

Comment:
Let's make sure there is no race condition effect here, as events to stage/run/remove, etc. tasks can come asynchronously.